frame/support/src/storage/mod.rs

```diff
@@ -733,6 +767,56 @@ impl<T> Iterator for PrefixIterator<T> {
 	}
 }
 
+/// Iterate over a prefix and decode raw_key into `T`.
+///
+/// If any decoding fails it skips it and continues to the next key.
+pub struct KeyPrefixIterator<T> {
+	prefix: Vec<u8>,
+	previous_key: Vec<u8>,
+	/// If true then value are removed while iterating
+	drain: bool,
```

Comment:
This is a normal way of doing this?

Reply:
We do similar for PrefixIterator, that would allow use to create `drain_keys` easily if we want.

Reply:
I actually spend a few minutes trying to see if we can merge this with `PrefixIterator`, the impls are pretty similar, only the decoding closure is different.

Reply:
@kianenigma seems hard to have the output be only keys or also keys and values.